src/mlpack/methods/ann/layer/atrous_convolution_impl.hpp

```diff
@@ -175,16 +188,23 @@ void AtrousConvolution<
 >::Backward(
     const arma::Mat<eT>&& /* input */, arma::Mat<eT>&& gy, arma::Mat<eT>&& g)
 {
-  arma::cube mappedError(gy.memptr(), outputWidth, outputHeight, outSize,
-      false, false);
+  arma::cube mappedError(gy.memptr(), outputWidth, outputHeight,
+      outSize * batchSize, false, false);
```

Comment:
Shouldn't it be possible to infer `batchSize` from the shape of `input`, `gy`, or `g`?

Reply:
Yes we can, but since we're anyways storing the `batchSize` in the forward pass, there's no need to call `input.n_cols` or `gy.n_cols` repeatedly.